`python311Packages.pillow-heif` build failure on x86_64-linux as of `846bf219`

## Issue description
Build of `python311Packages.pillow-heif` failed on x86_64-linux as of 846bf21967a372a24963697ed01eb643469101c6. This is currently breaking `python3Packages.wandb`.

```
error: builder for '/nix/store/fsvkwcnhp1f9n2dkz0w4q5qhibvs17g3-python3.11-pillow-heif-0.15.0.drv' failed with exit code 1;
       last 10 log lines:
       >             total_time_multiply_threads = perf_counter() - start_time_multiply_threads
       >             # decoding in multiply threads should be faster at least by 10%
       > >           assert total_time_one_thread > total_time_multiply_threads * 1.1
       > E           assert 0.5584241430005932 > (1.1149146390007445 * 1.1)
       >
       > /build/source/tests/options_test.py:113: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/options_test.py::test_decode_threads - assert 0.5584241430005932 > (1.1149146390007445 * 1.1)
       > ============= 1 failed, 899 passed, 8 skipped in 426.36s (0:07:06) =============
       > /nix/store/v099hqvw5z87423p4hz1vfhzaqa07dii-stdenv-linux/setup: line 1578: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/fsvkwcnhp1f9n2dkz0w4q5qhibvs17g3-python3.11-pillow-heif-0.15.0.drv'.
error: 1 dependencies of derivation '/nix/store/8yxgw86gasx4igzfp8kx6znq4wiwblb8-python3.11-imageio-2.34.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/39467ybqvcf32566r8nnpi0zz866f2sl-python3.11-wandb-0.16.0.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/8351806175)

python3Packages.wandb maintainers cc: @samuela
python311Packages.pillow-heif maintainers cc: @dali99
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 846bf21967a372a24963697ed01eb643469101c6
2. Run `nix-build -A python311Packages.pillow-heif`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.0-1016-azure, Ubuntu, 22.04.4 LTS (Jammy Jellyfish), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.16.1`
 - nixpkgs: `/nix/store/i45mwaiz0kpmqsda8ma0sxxn1p6wg4xk-source`
```

Contents of `~/.config/nixpkgs/config.nix`:
```
{ allowUnfree = true; cudaSupport = true; cudaCapabilities = ["8.0"]; }
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/8351806175) that created this issue.
- Internal tags: -7985494096423650684 -2077857451931816107

--- a/pkgs/development/python-modules/pillow-heif/default.nix
+++ b/pkgs/development/python-modules/pillow-heif/default.nix
@@ -75,7 +75,10 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
-  disabledTests = lib.optionals stdenv.isDarwin [
+  disabledTests = [
+    # Time based
+    "test_decode_threads"
+  ] ++ lib.optionals stdenv.isDarwin [
     # https://github.com/bigcat88/pillow_heif/issues/89
     # not reproducible in nixpkgs
     "test_opencv_crash"

